subxt/src/config/default_extrinsic_params.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+use super::{signed_extensions, ExtrinsicParams};
+use super::{Config, Header};
+
+/// The default [`super::ExtrinsicParams`] implementation understands common signed extensions
+/// and how to apply them to a given chain.
+pub type DefaultExtrinsicParams<T> = signed_extensions::AnyOf<
+    T,
+    (
+        signed_extensions::CheckSpecVersion,
+        signed_extensions::CheckTxVersion,
+        signed_extensions::CheckNonce,
+        signed_extensions::CheckGenesis<T>,
+        signed_extensions::CheckMortality<T>,
+        signed_extensions::ChargeAssetTxPayment,
+        signed_extensions::ChargeTransactionPayment,
+    ),
+>;
+
+/// A builder that outputs the set of [`super::ExtrinsicParams::OtherParams`] required for
+/// [`DefaultExtrinsicParams`]. This may expose methods that aren't applicable to the current
+/// chain; such values will simply be ignored if so.
+pub struct DefaultExtrinsicParamsBuilder<T: Config> {
+    mortality_checkpoint_hash: Option<T::Hash>,
+    mortality_checkpoint_number: u64,
+    mortality_period: u64,
+    tip: u128,
+    tip_of: u128,
+    tip_of_asset_id: Option<u32>,
+}
+
+impl<T: Config> Default for DefaultExtrinsicParamsBuilder<T> {
+    fn default() -> Self {
+        Self {
+            mortality_checkpoint_hash: None,
+            mortality_checkpoint_number: 0,
```

Comment:
default == immortal?

Reply:
could be cleaner to add `Option<Mortality>` instead of the three mortality fields

Reply:
Yup, good point, that's def cleaner!